app/ui/view_nano.c

```diff
@@ -183,7 +183,7 @@ zxerr_t h_review_update_data() {
             snprintf(viewdata.value, MAX_CHARS_PER_VALUE1_LINE, "Ok");
         } else {
 #if defined(APP_BLINDSIGN_MODE_ENABLED)
-            if (app_mode_blindsign()) {
+            if (app_mode_blindsign() && app_mode_blindsign_required()) {
```

Comment:
Sure I will zxlib major since this not have backward compatible.

Reply:
We could make it if we set  blindsign_required = 1 whenever blind signing mode is enabled. Then, you need to indicate which methods will not require the blind signing screens while this mode is enabled. I think this way is even more secure, as worst case we show screen when no needed. Instead of not showing them when it is indeed needed